### ISSUE TITLE
PSStyle means objects rendered in write-verbose look like normal output.

### ISSUE THREAD

**Issue by jhoneill** [PowerShell/PowerShell#16635]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

One of the advantages of the use of a different colour for write-verbose is that mixed with output the colour tells the user what is *output* and what is not. 
I find the green headings in 7.2 and 7.2.1 distracting so may profile has 
```
    $PSStyle.Formatting.TableHeader       = $null
    $PsStyle.Formatting.FormatAccent      = $null
```
Under 7.2.1 a sample like this 
```
write-verbose "about to kill some processes" -verbose ; 
Get-Process notepad | out-string | write-verbose  -Verbose
```
Produces the initial message in yellow, produces the column headers in yellow. 
Then at the end of the header line (in fact both header lines) `ESC[0M` is sent which turns the remaining text from yellow to white. This didn't seem to happen in 7.2 , I noticed it after the 7.2.1 update and after getting the sample below from some live code I spent some time trying to find what had caused the script to output objects instead of sending them to verbose.  

### Expected behavior

```console
All verbose output is printed in the colour selected for verbose
```

### Actual behavior

```console
Some isn't.
```

### Error details

```console
None
```

### Environment data

```powershell
Name                           Value[0m      <==   Note ansi rending bug with $psversionTable | clip
----                           -----[0m
PSVersion                      7.2.1
PSEdition                      Core
GitCommitId                    7.2.1
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0ΓÇª}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

![image](https://user-images.githubusercontent.com/16209593/146404050-26351f58-b7db-4e79-a4da-caa2a52219f4.png)

**Comment by 237dmitry** [comment]
`$PSStyle.OutputRendering = 'Host'` solves the output problem.

**Comment by jhoneill** [comment]
> `$PSStyle.OutputRendering = 'Host'` solves the output problem.

Well yes , but as we know form #16618  that's not the default and since being set to that is a security vulnerability it's hardly desirable.  Verbose should work correctly with the oob configuration, shouldn't it ?

**Comment by daxian-dbw** [comment]
Loop in @SteveL-MSFT to take a look.

**Comment by iSazonov** [comment]
> I find the green headings in 7.2 and 7.2.1 distracting 

Me too. This is a very sad situation. I don't know how long it takes to understand why color schemes in Windows are so carefully chosen.

**Comment by jhoneill** [comment]
> > I find the green headings in 7.2 and 7.2.1 distracting
> 
> Me too. This is a very sad situation. I don't know how long it takes to understand why color schemes in Windows are so carefully chosen.
> 
> It is in #16634 too.
 
#16634 was a user  thinking PowerShell's `ls` alias would output text like unix ls. So when he sent it to `select-string`, and got a a search through the _contents of all the files_ for the text "dll" instead of a search of the _file listing_  (a) It was unexpected and (b) there were some really long "lines" :-) 

Not sure if you mean #16618. I thought I posted to that one - maybe I was a bit forthright in my views and someone deleted it.  But that one means out-of-the-box PowerShell can't redirect to a file without getting Ansi-corruption.  `dir > file` has it ,  in the headers  `get-itemproperty` has it on every line. 

```
AuthenticationLevelOverride : [0m0
PSPath                      : [0mMicrosoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\SOFTWARE\Microsoft\Terminal Server Client\
PSParentPath                : [0mMicrosoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\SOFTWARE\Microsoft
PSChildName                 : [0mTerminal Server Client
PSDrive                     : [0mHKCU
PSProvider                  : [0mMicrosoft.PowerShell.Core\Registry
```

>  I don't know how long it takes to understand why color schemes in Windows are so carefully chosen.

What happens is sooner or later some dev or devs look at the monochrome text and think things would be a whole lot better if they were in colours or (heaven help us) flashing, or some such. They colour things in a way they like, but doesn't make any sense to anyone else - take a simple example, the experimental feature in Pwsh to apply colours to file listings says compressed files should be red. Red is usually used for "Danger" "Beware", "Not right", so why mark zips as worrying ? Simply because the guy who did it liked red.   For every person who likes the prettification, there's another who is driven mad by it.  No UI expert is ever involved. No one is able to tell them "You've made everyone less productive because they're spending more time reading and trying to understand what the visual differences mean".

**Comment by 237dmitry** [comment]
As a temporary solution, I have already added the `OutputRendering` (`PlainText/Ansi`) switch to my scripts. It's good that the mode can be switched on the fly.

**Comment by jhoneill** [comment]
> As a temporary solution, I have already added the `OutputRendering` (`PlainText/Ansi`) switch to my scripts. It's good that the mode can be switched on the fly.

Looks like no-one tested it in VS Code
![image](https://user-images.githubusercontent.com/16209593/146525178-97c2a017-d8b0-48f7-b1a6-86ae263bfabd.png)

**Comment by daxian-dbw** [comment]
@jhoneill Yeah, the VS Code uses a different host, and I think @andschwa hasn't get the chance to update the host appropriately to support `$PSStyle`, given that he has been busy with the stability issue with high priority. Do you mind open an issue in PSES repo to track this?

Same to the PowerShell sub-kernel in .NET Interactive, the host there hasn't been updated to properly support $PSStyle yet ...

**Comment by jhoneill** [comment]
> @jhoneill Yeah, the VS Code uses a different host, and I think @andschwa hasn't get the chance to update the host appropriately to support `$PSStyle`, given that he has been busy with the stability issue with high priority. Do you mind open an issue in PSES repo to track this?
> 
> Same to the PowerShell sub-kernel in .NET Interactive, the host there hasn't been updated to properly support $PSStyle yet ...

I started doing on in PSES but switched to #16643 here when I saw more than on one host was affected - just so I'm clear, although it's a change in PowerShell itself that caused it, the fix can only come from the people who produce the hosts?   I'm happy to open issues for those two, are there any more that need to be contacted  ?

**Comment by daxian-dbw** [comment]
> just so I'm clear, although it's a change in PowerShell itself that caused it, the fix can only come from the people who produce the hosts?

Unfortunately, yes. In PowerShell itself, the work that strips off the ANSI code is done in ConsoleHost, and that makes it necessary for other host to do something similar. But you made me start to think, maybe that's not the right design, maybe that should be done in formatting before sending the payload to host ...

**Comment by iSazonov** [comment]
> But you made me start to think, maybe that's not the right design, maybe that should be done in formatting before sending the payload to host

Oh, I talk about it all the time, but we all go around in "fix" circles... I would just do it in a separate optional module (MSFT team always talks about decoupling but doesn't do it itself 😸 ). Another thing is that the Engine _could_ have a generic coloring support to make it work efficiently and easily.

**Comment by daxian-dbw** [comment]
Actually, it turns out this is not a regression comparing with PS 5.1, even though it's still using `ConsoleColor`:

![image](https://user-images.githubusercontent.com/127450/168330919-4ba44834-95dd-44ce-9f8a-da9a30bcc2d2.png)

**Comment by jhoneill** [comment]
The issue described in the title is fixed in the the current preview of 7.3, together with the other issues of stray ansi sequences mentioned, and the matter of other hosts has its own issue.  Therefore, **I'll close this one one now.**  

If there will be another release of 7.2 before 7.3 is released it would be good to think about bringing these changes to 7.2, but I imagine all that would do is take effort away from getting 7.3 done.
